mcp/tests/staktrak/src/replay.ts

```diff
@@ -0,0 +1,1209 @@
+      (element as HTMLElement).style.transition = originalTransition;
+      element.classList.remove("replay-pulse");
+    }
+  }, 2000 / speedRef.current);
+}
+
+export function moveCursorToElement(
+  element: Element | null,
+  cursorRef: ElementRef,
+  statusRef: StatusRef
+): Promise<void> {
+  return new Promise((resolve) => {
+    if (
+      !element ||
+      !cursorRef.current ||
+      statusRef.current !== ReplayStatus.PLAYING
+    ) {
+      resolve();
+      return;
+    }
+
+    const rect = element.getBoundingClientRect();
+    const targetX = rect.left + rect.width / 2;
+    const targetY = rect.top + rect.height / 2;
+
+    cursorRef.current.style.display = "block";
+
+    element.scrollIntoView({ behavior: "smooth", block: "center" });
+
+    setTimeout(() => {
+      if (cursorRef.current) {
+        cursorRef.current.style.left = `${targetX}px`;
+        cursorRef.current.style.top = `${targetY}px`;
+      }
+
+      setTimeout(resolve, 800);
+    }, 400);
+  });
+}
+
+export function typeText(
+  element: HTMLInputElement,
+  value: string,
+  speedRef: SpeedRef,
+  statusRef: StatusRef,
+  isTypingRef: IsTypingRef,
+  registerTimeout: (timeout: NodeJS.Timeout) => NodeJS.Timeout
+): Promise<void> {
+  return new Promise((resolve) => {
+    if (statusRef.current !== ReplayStatus.PLAYING) {
+      resolve();
+      return;
+    }
+
+    isTypingRef.current = true;
+    element.focus();
+    element.value = "";
+
+    let index = 0;
+    const typeChar = () => {
+      if (statusRef.current !== ReplayStatus.PLAYING) {
+        isTypingRef.current = false;
+        resolve();
+        return;
+      }
+
+      if (index < value.length) {
+        element.value += value[index];
+        element.dispatchEvent(new Event("input", { bubbles: true }));
+        index++;
+        registerTimeout(setTimeout(typeChar, 100 / speedRef.current));
+      } else {
+        element.dispatchEvent(new Event("change", { bubbles: true }));
+        isTypingRef.current = false;
+        resolve();
+      }
+    };
+
+    typeChar();
+  });
+}
+
+export function selectOption(
+  element: HTMLSelectElement,
+  value: string,
+  speedRef: SpeedRef,
+  statusRef: StatusRef,
+  registerTimeout: (timeout: NodeJS.Timeout) => NodeJS.Timeout
+): Promise<void> {
+  return new Promise((resolve) => {
+    if (statusRef.current !== ReplayStatus.PLAYING) {
+      resolve();
+      return;
+    }
+
+    element.focus();
+
+    registerTimeout(
+      setTimeout(() => {
+        if (statusRef.current === ReplayStatus.PLAYING) {
+          element.value = value;
+          element.dispatchEvent(new Event("change", { bubbles: true }));
+        }
+        resolve();
+      }, 500 / speedRef.current)
+    );
+  });
+}
+
+export function createFallbackActions(): ReplayAction[] {
+  const actions: ReplayAction[] = [];
+  let timestamp = Date.now();
+
+  try {
+    const selectors = [
+      '[data-testid="staktrak-div"]',

```

Comment:
why are these hardcoded here? Don't put random fallbacks here. This is meant to run on any website